ls1intum/Artemis
# `Exam mode`: Fix score visibility for test exams

### Checklist
#### General
#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context
There are currently two issues when using test exams:
- After having the test exam open for a while, students can suddenly scores that they should not be able to see while solving the exam. They also no longer get updated results via the websocket.
- After submitting the exam, students should see their scores and test results. See #6207

### Description


### Steps for Testing
Prerequisites:
- 1 Test Exam, 1 real Exam
- 1 Student
- 1 Programming Exercise with all Test cases set to "After due Date"

Participate in both the real and test exam.
- In the real exam, at no point, the student should see scores or feedback.
- In the test exam, the students should not see the feedback during the exam, but they should see it after submitting theri test exam.

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage


### Screenshots
